pkg/collector/corechecks/servicediscovery/apm/detect.go

```diff
@@ -80,61 +82,42 @@ func isInjected(envs map[string]string) bool {
 	return false
 }
 
-func rubyDetector(_ []string, _ map[string]string) Instrumentation {
-	return None
+func getMaps(pid int) ([]byte, error) {
+	mapsPath := kernel.HostProc(strconv.Itoa(pid), "maps")
+	bytes, err := os.ReadFile(mapsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return bytes, nil
 }
 
-func pythonDetector(args []string, envs map[string]string) Instrumentation {
-	/*
-		Check for VIRTUAL_ENV env var
-			if it's there, use $VIRTUAL_ENV/lib/python{}/site-packages/ and see if ddtrace is inside
-			if so, return PROVIDED
-			if it's not there,
-				exec args[0] -c "import sys; print(':'.join(sys.path))"
-				split on :
-				for each part
-					see if it ends in site-packages
-					if so, check if ddtrace is inside
-						if so, return PROVIDED
-			return NONE
-	*/
-	if path, ok := envs["VIRTUAL_ENV"]; ok {
-		venv := os.DirFS(path)
-		libContents, err := fs.ReadDir(venv, "lib")
-		if err == nil {
-			for _, v := range libContents {
-				if strings.HasPrefix(v.Name(), "python") && v.IsDir() {
-					tracedir, err := fs.Stat(venv, "lib/"+v.Name()+"/site-packages/ddtrace")
-					if err == nil && tracedir.IsDir() {
-						return Provided
-					}
-				}
-			}
-		}
-		// the virtual env didn't have ddtrace, can exit
-		return None
+func pythonDetectorFromMaps(maps []byte) Instrumentation {
+	idx := bytes.Index(maps, []byte("/site-packages/ddtrace/"))
+	if idx != -1 {
+		return Provided
 	}
-	// slow option...
-	results, err := exec.Command(args[0], `-c`, `"import sys; print(':'.join(sys.path))"`).Output()
+
+	return None
+}
+
+// pythonDetector detects the use of the ddtrace package in the process. Since
+// the ddtrace package uses native libraries, the paths of these libraries will
+// show up in /proc/$PID/maps.
+//
+// For example:
+// 7aef453fc000-7aef453ff000 rw-p 0004c000 fc:06 7895473  /home/foo/.local/lib/python3.10/site-packages/ddtrace/internal/_encoding.cpython-310-x86_64-linux-gnu.so
+// 7aef45400000-7aef45459000 r--p 00000000 fc:06 7895588  /home/foo/.local/lib/python3.10/site-packages/ddtrace/internal/datadog/profiling/libdd_wrapper.so
+func pythonDetector(pid int, _ []string, _ map[string]string) Instrumentation {
```

Comment:
nit - I think we should create a struct to be passed to the different detectors.
I assume that overtime we will add more fields and attributes to the signature of the detector as different detectors will need additional (unique?) information.

If you agree, then let's create a jira ticket to follow that